app/models/shift.rb

```diff
@@ -2,7 +2,10 @@ class Shift < ApplicationRecord
   belongs_to :user
   belongs_to :schedule
   
+  validates :user, presence: true
   validates :schedule, presence: true
+  validates :shift_start, presence: true
+  validates :shift_end, presence: true
```

Comment:
Outside the scope of this PR given how large it is already, but this could do with ensuring that the end date is after or equal to the start, so you don't end up with an inverted shift.